### ISSUE TITLE
FFmpeg won't install

### ISSUE THREAD

**Issue by HunterAmaku** [axcore/tartube#371]
****DEVELOPER EDIT**: On MS Windows, FFmpeg cannot be installed because of bad security certificates in the MSYS2 repository. You can try [this solution](https://github.com/axcore/tartube/issues/371#issuecomment-1024896301) or wait for the certificates to be updated.**

OS: Windows 11
Tartube Version: v2.3.332

Whenever I go to install FFmpeg, it sticks at:

checking keyring...
checking package integrity...

for a minute or so, then it says:

:: File /var/cache/pacman/pkg/mingw-w64-x86_64-eigen3-3.3.9-1-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-openblas-0.3.13-2-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-suitesparse-5.8.1-1-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-ceres-solver-2.0.0-2-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-lcms2-2.12-1-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-libraw-0.20.2-2-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-SDL2-2.0.14-2-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-pugixml-1.11.4-1-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-jansson-2.13.1-1-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-lz4-1.9.3-1-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-libarchive-3.5.1-1-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-libbluray-1.2.1-1-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-libsoxr-0.1.3-2-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-libvorbis-1.3.7-1-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-srt-1.4.2-1-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-ffms2-2.40-1-any.pkg.tar.zst is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
Errors occurred, no packages were upgraded.
error: mingw-w64-x86_64-eigen3: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: mingw-w64-x86_64-openblas: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: mingw-w64-x86_64-suitesparse: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: mingw-w64-x86_64-ceres-solver: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: mingw-w64-x86_64-lcms2: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: mingw-w64-x86_64-libraw: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: mingw-w64-x86_64-SDL2: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: mingw-w64-x86_64-pugixml: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: mingw-w64-x86_64-jansson: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: mingw-w64-x86_64-lz4: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: mingw-w64-x86_64-libarchive: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: mingw-w64-x86_64-libbluray: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: mingw-w64-x86_64-libsoxr: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: mingw-w64-x86_64-libvorbis: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: mingw-w64-x86_64-srt: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: mingw-w64-x86_64-ffms2: signature from "David Macek <david.macek.0@gmail.com>" is unknown trust
error: failed to commit transaction (invalid or corrupted package)
Update operation finished

**Comment by NotErikUden** [comment]
I am having the same issue. Made a thread about it on Reddit. 
https://www.reddit.com/r/youtubedl/comments/scmpcb/
No solutions to me so far.
Pinged the Author of Tartube! I am using the same version as you, but I am on Windows 10.

**Comment by axcore** [comment]
I have no internet at the moment, and also no computer with MS Windows installed, so I can't do any testing for this issue.

If FFmpeg installation was working a week ago and is now not working, then it's most likely to be a problem with the servers that host the FFmpeg packages. Nothing that can be done about it, but to wait until it's fixed. 

You might try deleting Tartube's installation folder (on MS Windows, the default location is **C:\Users\YOURNAME\AppData\Local\Tartube**), and then reinstalling Tartube. That will clear the packages previously downloaded from the server, in case they have corrupted each other.

**Comment by datlaunchystark** [comment]
I have the same issue too.  Maybe we'll have to wait for the devs to fix this..

**Comment by h3ntaih3ro** [comment]
> I have the same issue too. One work around that I did end up doing, while not fully recommended, I edited the file pacman.conf in AppData\Local\Tartube\msys64\etc and chaged the "SigLevel" from 'Required' to 'Never'. Maybe we'll have to wait for the devs to fix this and include the package signatures and maybe a terminal mode would be great too.
> 
> If you need the command line for Tartube's backend to do some things you can run AppData\Local\Tartube\msys64\mingw64.exe and attempt to update or sign things there.

This fix worked for me. Thanks a lot!

**Comment by axcore** [comment]
The repository from which Tartube installs ffmpeg is [here](https://github.com/msys2/MINGW-packages), so I suppose the thing to do would be to open an issue there, complaining that ffmpeg is uninstallable from mingw64.exe using this command:

pacman -S mingw-w64-x86_64-ffmpeg

(I won't have internet until late February, so don't wait for me to do it).

It would be simple to update the Tartube installer to change pacman.conf's SigLevel, but I am reluctant to do that: in case someone has tampered with the ffmpeg repository.

**Comment by axcore** [comment]
>maybe a terminal mode would be great too.

I have considered in the past, but it would be quite a lot of work, and I can't think of an advantage over (say) a simple python script that started multiple instances of youtube-dl, one for each channel to be downloaded. But maybe someone can think of a compelling use case.

For anyone who doesn't know, Tartube can be minimised to the tray (Edit > System preferences... > Windows > Tray > Close to the tray), which is almost the same thing.

**Comment by axcore** [comment]
You can try v2.3.367 which was released today. I was able to install FFmpeg using it, but only after a couple of failed attempts. (The failure is probably the result of WIFI issues).

**Comment by metapea** [comment]
> You can try v2.3.367 which was released today. I was able to install FFmpeg using it, but only after a couple of failed attempts. (The failure is probably the result of WIFI issues).

The new version failed for me:
```
installing mingw-w64-x86_64-docbook-xml...
/tmp/alpm_FpvM3t/.INSTALL: line 10: mingw64/bin/xmlcatalog: No such file or
directory
/tmp/alpm_FpvM3t/.INSTALL: line 12: mingw64/bin/xmlcatalog: No such file or
directory
/tmp/alpm_FpvM3t/.INSTALL: line 16: mingw64/bin/xmlcatalog: No such file or
directory
/tmp/alpm_FpvM3t/.INSTALL: line 20: mingw64/bin/xmlcatalog: No such file or
directory
/tmp/alpm_FpvM3t/.INSTALL: line 24: mingw64/bin/xmlcatalog: No such file or
directory
installing mingw-w64-x86_64-docbook-xsl...
/tmp/alpm_U8KQe7/.INSTALL: line 5: /mingw64/bin/xmlcatalog: No such file or
directory
/tmp/alpm_U8KQe7/.INSTALL: line 5: /mingw64/bin/xmlcatalog: No such file or
directory
/tmp/alpm_U8KQe7/.INSTALL: line 5: /mingw64/bin/xmlcatalog: No such file or
directory
/tmp/alpm_U8KQe7/.INSTALL: line 5: /mingw64/bin/xmlcatalog: No such file or
directory
/tmp/alpm_U8KQe7/.INSTALL: line 5: /mingw64/bin/xmlcatalog: No such file or
directory
/tmp/alpm_U8KQe7/.INSTALL: line 5: /mingw64/bin/xmlcatalog: No such file or
directory
/tmp/alpm_U8KQe7/.INSTALL: line 5: /mingw64/bin/xmlcatalog: No such file or
directory
/tmp/alpm_U8KQe7/.INSTALL: line 5: /mingw64/bin/xmlcatalog: No such file or
directory
/tmp/alpm_U8KQe7/.INSTALL: line 5: /mingw64/bin/xmlcatalog: No such file or
directory

```

`error: command (/usr/bin/bash /usr/bin/bash -c . /tmp/alpm_FpvM3t/.INSTALL;
post_install 1~4.5-2 ) failed to execute correctly
error: command (/usr/bin/bash /usr/bin/bash -c . /tmp/alpm_U8KQe7/.INSTALL;
post_install 1.79.2-7 ) failed to execute correctly
Update operation finished
Installation failed`

**Comment by NotErikUden** [comment]
> You can try v2.3.367 which was released today. I was able to install FFmpeg using it, but only after a couple of failed attempts. (The failure is probably the result of WIFI issues).

Am currently on my third try, sadly it does not seem to work for me either. No worries, however, I have a functioning installation on a different device, and have been downloading lots and lots of videos from there! Thanks to your recent comment regarding historical statistics, I even know how many! Haha.

**Comment by NotErikUden** [comment]
(works for me!)

**Comment by Flavescent** [comment]
I get the same error when attempting to install ffmpeg in Tartube on Win10 today.

**Comment by NotErikUden** [comment]
Try to do what I did to fix it!

**Comment by heywolf15** [comment]
What ends up working for me was, re-install Tartube, make dummy folder for new database, successfully installing ffmpeg in the third try, in the third try it successfully installs all the dependencies with one error in the end, fourth try just to see there's no error.
Finally, deleting dummy database account and change it to my old database.

**Comment by axcore** [comment]
MSYS2's ffmpeg repository has been updated. I tested it today, I installed ffmpeg three times without any problems. If any of you have time to confirm this, it would be useful. (Make sure you are using v2.3.367 or it probably won't work.)

**Comment by axcore** [comment]
@metapea Here is the missing file, https://github.com/axcore/tartube-test.

Copy the xmlcatalog file into **..\msys64\mingw64\bin\,** for example **C:\Users\YOURNAME\AppData\Local\Tartube\msys64\mingw64\bin\**

**Comment by metapea** [comment]
> @metapea Here is the missing file, https://github.com/axcore/tartube-test.
> 
> Copy the xmlcatalog file into **..\msys64\mingw64\bin,** for example *_C:\Users\YOURNAME\AppData\Local\Tartube\msys64\mingw64\bin*_

I tried pacman -S mingw-w64-x86_64-docbook-xsl and pacman -S mingw-w64-x86_64-docbook-xml again:
```
Failed to remove entry ../../share/xml/docbook/xsl-stylesheets-1.79.2
Failed to remove entry http://docbook.sourceforge.net/release/xsl/
Failed to remove entry http://docbook.sourceforge.net/release/xsl-ns/
/tmp/alpm_HSiY65/.INSTALL: line 41: rm: command not found
/tmp/alpm_HSiY65/.INSTALL: line 42: vercmp: command not found
/tmp/alpm_HSiY65/.INSTALL: line 42: ((: >= 0 : syntax error: operand expected (error token is ">= 0 ")
add command failed
add command failed
add command failed
add command failed
add command failed
add command failed
add command failed
add command failed
error: command (/usr/bin/bash /usr/bin/bash -c . /tmp/alpm_HSiY65/.INSTALL; post_upgrade 1.79.2-7 1.79.2-7 ) failed to execute correctly
```

and:
```
/tmp/alpm_unq9o4/.INSTALL: line 33: vercmp: command not found
/tmp/alpm_unq9o4/.INSTALL: line 33: [: -lt: unary operator expected
(1/1) reinstalling mingw-w64-x86_64-docbook-xml    [#####################] 100%
/tmp/alpm_HVMhOI/.INSTALL: line 40: vercmp: command not found
/tmp/alpm_HVMhOI/.INSTALL: line 40: [: -ge: unary operator expected
add command failed
add command failed
add command failed
add command failed
error: command (/usr/bin/bash /usr/bin/bash -c . /tmp/alpm_HVMhOI/.INSTALL; post_upgrade 1~4.5-2 1~4.5-2 ) failed to execute correctly
```

**Comment by axcore** [comment]
Ok, so `../Tartube/msys64/usr/bin/rm` and` ../Tartube/msys64/usr/bin/vercmp` are missing from the installer. I have added them to https://github.com/axcore/tartube-test.

But I think this might not help you. The question is not, "how can I install docbook" but "how can I install FFmpeg". Installing docbook separately may be a different process from installing docbook as a dependency of FFmpeg.

Perhaps you could try installing FFmpeg on a separate MSYS2 system, at least then we will know if the problem is my installer, or not:

- Download and install [MSYS2](https://www.msys2.org/)
- Open the terminal window, **C:\\msys64\\mingw64.exe**
- In the new terminal window, run **pacman -Syu**
- When the terminal wants to close, close it
- Run the terminal window again
- Type **pacman -SU**
- Type **pacman -S mingw-w64-x86_64-ffmpeg**

**Comment by metapea** [comment]
> * Download and install [MSYS2](https://www.msys2.org/)
> * Open the terminal window, **C:\msys64\mingw64.exe**
> * In the new terminal window, run **pacman -Syu**
> * When the terminal wants to close, close it
> * Run the terminal window again
> * Type **pacman -SU**
> * Type **pacman -S mingw-w64-x86_64-ffmpeg**

When i typed **pacman -SU** it give me:
>_error: only one operation may be used at a time_

But the whole thing installed fine when i typed "pacman -S mingw-w64-x86_64-ffmpeg" afterwords.

**Comment by axcore** [comment]
> When i typed **pacman -SU** it give me:

Apologies, should have been a lower-case U: **pacman -Su**

>But the whole thing installed fine when i typed "pacman -S mingw-w64-x86_64-ffmpeg" afterwords.

Great. I have to do a new release within the next day or two, due to other problems with the installer, so you can try that.

Sorry, I have no idea why the docbook stuff doesn't install.

**Comment by metapea** [comment]
> > When i typed **pacman -SU** it give me:
> 
> Apologies, should have been a lower-case U: **pacman -Su**
> 
> > But the whole thing installed fine when i typed "pacman -S mingw-w64-x86_64-ffmpeg" afterwords.
> 
> Great. I have to do a new release within the next day or two, due to other problems with the installer, so you can try that.
> 
> Sorry, I have no idea why the docbook stuff doesn't install.

It's working.
![Screenshot_339](https://user-images.githubusercontent.com/35351174/159183011-0f6c41fc-14dd-4079-a088-40d59733bf0a.png)

**Comment by axcore** [comment]
>it's working on tartube.

Do you mean that FFmpeg can be installed using the current release of Tartube, v2.3.447 (for MS WIndows)?

**Comment by metapea** [comment]
yep, the bug should be fixed hopefully

**Comment by welpdx** [comment]
I get the exact same error during the start up installation process after installing yt-dlp

<details><summary>log</summary>
<p>

```cmd
Starting update operation, installing FFmpeg
pacman -S mingw-w64-x86_64-ffmpeg --noconfirm
resolving dependencies...
looking for conflicting packages...



Packages (96) mingw-w64-x86_64-SDL2-2.0.20-1 mingw-w64-x86_64-aom-3.3.0-2
mingw-w64-x86_64-asciidoc-10.1.4-1 mingw-w64-x86_64-boost-1.78.0-3
mingw-w64-x86_64-celt-0.11.3-4 mingw-w64-x86_64-ceres-solver-2.0.0-3
mingw-w64-x86_64-cppunit-1.15.1-2
mingw-w64-x86_64-crt-git-9.0.0.6448.b03cbfb95-1 mingw-w64-x86_64-curl-7.82.0-1
mingw-w64-x86_64-dav1d-0.9.2-1 mingw-w64-x86_64-docbook-xml-1~4.5-2
mingw-w64-x86_64-docbook-xsl-1.79.2-7 mingw-w64-x86_64-eigen3-3.4.0-1
mingw-w64-x86_64-ffms2-2.40-1 mingw-w64-x86_64-freeglut-3.2.2-1
mingw-w64-x86_64-freeimage-3.18.0-7 mingw-w64-x86_64-frei0r-plugins-1.7.0-3
mingw-w64-x86_64-gavl-1.4.0-1 mingw-w64-x86_64-gcc-libgfortran-11.2.0-10
mingw-w64-x86_64-gflags-2.2.2-4 mingw-w64-x86_64-glog-0.5.0-3
mingw-w64-x86_64-glsl-optimizer-git-1~r66913.d78c3d2f24-1
mingw-w64-x86_64-gnutls-3.7.4-1 mingw-w64-x86_64-gsm-1.0.19-1
mingw-w64-x86_64-hdf5-1.12.1-2
mingw-w64-x86_64-headers-git-9.0.0.6448.b03cbfb95-1
mingw-w64-x86_64-hlsl2glsl-git-1~r848.14dd189-1 mingw-w64-x86_64-icu-69.1-1
mingw-w64-x86_64-ilmbase-2.5.7-6 mingw-w64-x86_64-intel-tbb-1~2020.3-1
mingw-w64-x86_64-jansson-2.14-2 mingw-w64-x86_64-jasper-2.0.33-1
mingw-w64-x86_64-jemalloc-5.2.1-2 mingw-w64-x86_64-jxrlib-1.1-3
mingw-w64-x86_64-l-smash-2.14.5-2 mingw-w64-x86_64-lame-3.100-1
mingw-w64-x86_64-lcms2-2.13.1-2 mingw-w64-x86_64-leptonica-1.81.1-1
mingw-w64-x86_64-libaec-1.0.6-2 mingw-w64-x86_64-libarchive-3.6.0-1
mingw-w64-x86_64-libass-0.15.2-3 mingw-w64-x86_64-libb2-0.98.1-2
mingw-w64-x86_64-libbluray-1.3.0-2 mingw-w64-x86_64-libcaca-0.99.beta20-1
mingw-w64-x86_64-libexif-0.6.24-2 mingw-w64-x86_64-libgcrypt-1.10.0-1
mingw-w64-x86_64-libgme-0.6.3-3 mingw-w64-x86_64-libgpg-error-1.44-1
mingw-w64-x86_64-libidn2-2.3.1-1 mingw-w64-x86_64-libmfx-1.25-1
mingw-w64-x86_64-libmodplug-0.8.9.0-4 mingw-w64-x86_64-libmysofa-1.2.1-1
mingw-w64-x86_64-libogg-1.3.5-1 mingw-w64-x86_64-libpsl-0.21.1-4
mingw-w64-x86_64-libraw-0.20.2-5 mingw-w64-x86_64-libsoxr-0.1.3-3
mingw-w64-x86_64-libssh-0.9.6-2 mingw-w64-x86_64-libtheora-1.1.1-6
mingw-w64-x86_64-libunistring-0.9.10-4 mingw-w64-x86_64-libunwind-13.0.1-1
mingw-w64-x86_64-libvorbis-1.3.7-1 mingw-w64-x86_64-libvpx-1.11.0-2
mingw-w64-x86_64-libxslt-1.1.35-1 mingw-w64-x86_64-lz4-1.9.3-1
mingw-w64-x86_64-metis-5.1.0-4 mingw-w64-x86_64-nettle-3.7.3-3
mingw-w64-x86_64-nghttp2-1.47.0-1 mingw-w64-x86_64-ogre3d-1.12.6-4
mingw-w64-x86_64-openal-1.21.1-4 mingw-w64-x86_64-openblas-0.3.20-1
mingw-w64-x86_64-opencore-amr-0.1.5-1 mingw-w64-x86_64-opencv-4.5.2-4
mingw-w64-x86_64-openexr-2.5.7-6 mingw-w64-x86_64-openjpeg2-2.4.0-3
mingw-w64-x86_64-opus-1.3.1-4 mingw-w64-x86_64-protobuf-3.16.0-3
mingw-w64-x86_64-pugixml-1.12-2 mingw-w64-x86_64-rav1e-0.4.1-1
mingw-w64-x86_64-rtmpdump-git-r514.c5f04a5-4 mingw-w64-x86_64-speex-1.2.0-1
mingw-w64-x86_64-speexdsp-1.2.0-1 mingw-w64-x86_64-srt-1.4.4-1
mingw-w64-x86_64-suitesparse-5.10.1-2 mingw-w64-x86_64-svt-av1-0.9.1-1
mingw-w64-x86_64-tesseract-ocr-5.1.0-1 mingw-w64-x86_64-tinyxml-2.6.2-5
mingw-w64-x86_64-vid.stab-1.1-1 mingw-w64-x86_64-vulkan-headers-1.3.207-1
mingw-w64-x86_64-vulkan-loader-1.3.207-1
mingw-w64-x86_64-winpthreads-git-9.0.0.6448.b03cbfb95-1
mingw-w64-x86_64-x264-0.161.r3039.544c61f-2 mingw-w64-x86_64-x265-3.5-1
mingw-w64-x86_64-xvidcore-1.3.7-2 mingw-w64-x86_64-zimg-3.0.3-1
mingw-w64-x86_64-zziplib-0.13.72-3 mingw-w64-x86_64-ffmpeg-4.4.1-11
Total Download Size: 280.41 MiB
Total Installed Size: 1551.40 MiB
:: Proceed with installation? [Y/n]
:: Retrieving packages...
mingw-w64-x86_64-SDL2-2.0.20-1-any downloading...
mingw-w64-x86_64-aom-3.3.0-2-any downloading...
mingw-w64-x86_64-asciidoc-10.1.4-1-any downloading...
mingw-w64-x86_64-boost-1.78.0-3-any downloading...
mingw-w64-x86_64-celt-0.11.3-4-any downloading...
mingw-w64-x86_64-ceres-solver-2.0.0-3-any downloading...
mingw-w64-x86_64-cppunit-1.15.1-2-any downloading...
mingw-w64-x86_64-crt-git-9.0.0.6448.b03cbfb95-1-any downloading...
mingw-w64-x86_64-curl-7.82.0-1-any downloading...
mingw-w64-x86_64-dav1d-0.9.2-1-any downloading...
mingw-w64-x86_64-docbook-xml-1~4.5-2-any downloading...
mingw-w64-x86_64-docbook-xsl-1.79.2-7-any downloading...
mingw-w64-x86_64-eigen3-3.4.0-1-any downloading...
mingw-w64-x86_64-ffmpeg-4.4.1-11-any downloading...
mingw-w64-x86_64-ffms2-2.40-1-any downloading...
mingw-w64-x86_64-freeglut-3.2.2-1-any downloading...
mingw-w64-x86_64-freeimage-3.18.0-7-any downloading...
mingw-w64-x86_64-frei0r-plugins-1.7.0-3-any downloading...
mingw-w64-x86_64-gavl-1.4.0-1-any downloading...
mingw-w64-x86_64-gcc-libgfortran-11.2.0-10-any downloading...
mingw-w64-x86_64-gflags-2.2.2-4-any downloading...
mingw-w64-x86_64-glog-0.5.0-3-any downloading...
mingw-w64-x86_64-glsl-optimizer-git-1~r66913.d78c3d2f24-1-any downloading...
mingw-w64-x86_64-gnutls-3.7.4-1-any downloading...
mingw-w64-x86_64-gsm-1.0.19-1-any downloading...
mingw-w64-x86_64-hdf5-1.12.1-2-any downloading...
mingw-w64-x86_64-headers-git-9.0.0.6448.b03cbfb95-1-any downloading...
mingw-w64-x86_64-hlsl2glsl-git-1~r848.14dd189-1-any downloading...
mingw-w64-x86_64-icu-69.1-1-any downloading...
mingw-w64-x86_64-ilmbase-2.5.7-6-any downloading...
mingw-w64-x86_64-intel-tbb-1~2020.3-1-any downloading...
mingw-w64-x86_64-jansson-2.14-2-any downloading...
mingw-w64-x86_64-jasper-2.0.33-1-any downloading...
mingw-w64-x86_64-jemalloc-5.2.1-2-any downloading...
mingw-w64-x86_64-jxrlib-1.1-3-any downloading...
mingw-w64-x86_64-l-smash-2.14.5-2-any downloading...
mingw-w64-x86_64-lame-3.100-1-any downloading...
mingw-w64-x86_64-lcms2-2.13.1-2-any downloading...
mingw-w64-x86_64-leptonica-1.81.1-1-any downloading...
mingw-w64-x86_64-libaec-1.0.6-2-any downloading...
mingw-w64-x86_64-libarchive-3.6.0-1-any downloading...
mingw-w64-x86_64-libass-0.15.2-3-any downloading...
mingw-w64-x86_64-libb2-0.98.1-2-any downloading...
mingw-w64-x86_64-libbluray-1.3.0-2-any downloading...
mingw-w64-x86_64-libcaca-0.99.beta20-1-any downloading...
mingw-w64-x86_64-libexif-0.6.24-2-any downloading...
mingw-w64-x86_64-libgcrypt-1.10.0-1-any downloading...
mingw-w64-x86_64-libgme-0.6.3-3-any downloading...
mingw-w64-x86_64-libgpg-error-1.44-1-any downloading...
mingw-w64-x86_64-libidn2-2.3.1-1-any downloading...
mingw-w64-x86_64-libmfx-1.25-1-any downloading...
mingw-w64-x86_64-libmodplug-0.8.9.0-4-any downloading...
mingw-w64-x86_64-libmysofa-1.2.1-1-any downloading...
mingw-w64-x86_64-libogg-1.3.5-1-any downloading...
mingw-w64-x86_64-libpsl-0.21.1-4-any downloading...
mingw-w64-x86_64-libraw-0.20.2-5-any downloading...
mingw-w64-x86_64-libsoxr-0.1.3-3-any downloading...
mingw-w64-x86_64-libssh-0.9.6-2-any downloading...
mingw-w64-x86_64-libtheora-1.1.1-6-any downloading...
mingw-w64-x86_64-libunistring-0.9.10-4-any downloading...
mingw-w64-x86_64-libunwind-13.0.1-1-any downloading...
mingw-w64-x86_64-libvorbis-1.3.7-1-any downloading...
mingw-w64-x86_64-libvpx-1.11.0-2-any downloading...
mingw-w64-x86_64-libxslt-1.1.35-1-any downloading...
mingw-w64-x86_64-lz4-1.9.3-1-any downloading...
mingw-w64-x86_64-metis-5.1.0-4-any downloading...
mingw-w64-x86_64-nettle-3.7.3-3-any downloading...
mingw-w64-x86_64-nghttp2-1.47.0-1-any downloading...
mingw-w64-x86_64-ogre3d-1.12.6-4-any downloading...
mingw-w64-x86_64-openal-1.21.1-4-any downloading...
mingw-w64-x86_64-openblas-0.3.20-1-any downloading...
mingw-w64-x86_64-opencore-amr-0.1.5-1-any downloading...
mingw-w64-x86_64-opencv-4.5.2-4-any downloading...
mingw-w64-x86_64-openexr-2.5.7-6-any downloading...
mingw-w64-x86_64-openjpeg2-2.4.0-3-any downloading...
mingw-w64-x86_64-opus-1.3.1-4-any downloading...
mingw-w64-x86_64-protobuf-3.16.0-3-any downloading...
mingw-w64-x86_64-pugixml-1.12-2-any downloading...
mingw-w64-x86_64-rav1e-0.4.1-1-any downloading...
mingw-w64-x86_64-rtmpdump-git-r514.c5f04a5-4-any downloading...
mingw-w64-x86_64-speex-1.2.0-1-any downloading...
mingw-w64-x86_64-speexdsp-1.2.0-1-any downloading...
mingw-w64-x86_64-srt-1.4.4-1-any downloading...
mingw-w64-x86_64-suitesparse-5.10.1-2-any downloading...
mingw-w64-x86_64-svt-av1-0.9.1-1-any downloading...
mingw-w64-x86_64-tesseract-ocr-5.1.0-1-any downloading...
mingw-w64-x86_64-tinyxml-2.6.2-5-any downloading...
mingw-w64-x86_64-vid.stab-1.1-1-any downloading...
mingw-w64-x86_64-vulkan-headers-1.3.207-1-any downloading...
mingw-w64-x86_64-vulkan-loader-1.3.207-1-any downloading...
mingw-w64-x86_64-winpthreads-git-9.0.0.6448.b03cbfb95-1-any downloading...
mingw-w64-x86_64-x264-0.161.r3039.544c61f-2-any downloading...
mingw-w64-x86_64-x265-3.5-1-any downloading...
mingw-w64-x86_64-xvidcore-1.3.7-2-any downloading...
mingw-w64-x86_64-zimg-3.0.3-1-any downloading...
mingw-w64-x86_64-zziplib-0.13.72-3-any downloading...
checking keyring...
checking package integrity...
error: mingw-w64-x86_64-lz4: signature from "David Macek
<david.macek.0@gmail.com>" is unknown trust
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-lz4-1.9.3-1-any.pkg.tar.zst is
corrupted (invalid or corrupted package (PGP signature)).
error: mingw-w64-x86_64-libvorbis: signature from "David Macek
<david.macek.0@gmail.com>" is unknown trust
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-libvorbis-1.3.7-1-any.pkg.tar.zst
is corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
error: mingw-w64-x86_64-ffms2: signature from "David Macek
<david.macek.0@gmail.com>" is unknown trust
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-ffms2-2.40-1-any.pkg.tar.zst is
corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
error: failed to commit transaction (invalid or corrupted package)
Do you want to delete it? [Y/n]
Errors occurred, no packages were upgraded.
Update operation finished
Installation failed

```

</p>
</details>


Same error on both portable and installer versions. Win 10 Pro 21H1 19043.1766
<details><summary>SigLevel = Never </summary>
<p>

```cmd
Starting update operation, installing FFmpeg
pacman -S mingw-w64-x86_64-ffmpeg --noconfirm
resolving dependencies...
looking for conflicting packages...



Packages (96) mingw-w64-x86_64-SDL2-2.0.20-1 mingw-w64-x86_64-aom-3.3.0-2
mingw-w64-x86_64-asciidoc-10.1.4-1 mingw-w64-x86_64-boost-1.78.0-3
mingw-w64-x86_64-celt-0.11.3-4 mingw-w64-x86_64-ceres-solver-2.0.0-3
mingw-w64-x86_64-cppunit-1.15.1-2
mingw-w64-x86_64-crt-git-9.0.0.6448.b03cbfb95-1 mingw-w64-x86_64-curl-7.82.0-1
mingw-w64-x86_64-dav1d-0.9.2-1 mingw-w64-x86_64-docbook-xml-1~4.5-2
mingw-w64-x86_64-docbook-xsl-1.79.2-7 mingw-w64-x86_64-eigen3-3.4.0-1
mingw-w64-x86_64-ffms2-2.40-1 mingw-w64-x86_64-freeglut-3.2.2-1
mingw-w64-x86_64-freeimage-3.18.0-7 mingw-w64-x86_64-frei0r-plugins-1.7.0-3
mingw-w64-x86_64-gavl-1.4.0-1 mingw-w64-x86_64-gcc-libgfortran-11.2.0-10
mingw-w64-x86_64-gflags-2.2.2-4 mingw-w64-x86_64-glog-0.5.0-3
mingw-w64-x86_64-glsl-optimizer-git-1~r66913.d78c3d2f24-1
mingw-w64-x86_64-gnutls-3.7.4-1 mingw-w64-x86_64-gsm-1.0.19-1
mingw-w64-x86_64-hdf5-1.12.1-2
mingw-w64-x86_64-headers-git-9.0.0.6448.b03cbfb95-1
mingw-w64-x86_64-hlsl2glsl-git-1~r848.14dd189-1 mingw-w64-x86_64-icu-69.1-1
mingw-w64-x86_64-ilmbase-2.5.7-6 mingw-w64-x86_64-intel-tbb-1~2020.3-1
mingw-w64-x86_64-jansson-2.14-2 mingw-w64-x86_64-jasper-2.0.33-1
mingw-w64-x86_64-jemalloc-5.2.1-2 mingw-w64-x86_64-jxrlib-1.1-3
mingw-w64-x86_64-l-smash-2.14.5-2 mingw-w64-x86_64-lame-3.100-1
mingw-w64-x86_64-lcms2-2.13.1-2 mingw-w64-x86_64-leptonica-1.81.1-1
mingw-w64-x86_64-libaec-1.0.6-2 mingw-w64-x86_64-libarchive-3.6.0-1
mingw-w64-x86_64-libass-0.15.2-3 mingw-w64-x86_64-libb2-0.98.1-2
mingw-w64-x86_64-libbluray-1.3.0-2 mingw-w64-x86_64-libcaca-0.99.beta20-1
mingw-w64-x86_64-libexif-0.6.24-2 mingw-w64-x86_64-libgcrypt-1.10.0-1
mingw-w64-x86_64-libgme-0.6.3-3 mingw-w64-x86_64-libgpg-error-1.44-1
mingw-w64-x86_64-libidn2-2.3.1-1 mingw-w64-x86_64-libmfx-1.25-1
mingw-w64-x86_64-libmodplug-0.8.9.0-4 mingw-w64-x86_64-libmysofa-1.2.1-1
mingw-w64-x86_64-libogg-1.3.5-1 mingw-w64-x86_64-libpsl-0.21.1-4
mingw-w64-x86_64-libraw-0.20.2-5 mingw-w64-x86_64-libsoxr-0.1.3-3
mingw-w64-x86_64-libssh-0.9.6-2 mingw-w64-x86_64-libtheora-1.1.1-6
mingw-w64-x86_64-libunistring-0.9.10-4 mingw-w64-x86_64-libunwind-13.0.1-1
mingw-w64-x86_64-libvorbis-1.3.7-1 mingw-w64-x86_64-libvpx-1.11.0-2
mingw-w64-x86_64-libxslt-1.1.35-1 mingw-w64-x86_64-lz4-1.9.3-1
mingw-w64-x86_64-metis-5.1.0-4 mingw-w64-x86_64-nettle-3.7.3-3
mingw-w64-x86_64-nghttp2-1.47.0-1 mingw-w64-x86_64-ogre3d-1.12.6-4
mingw-w64-x86_64-openal-1.21.1-4 mingw-w64-x86_64-openblas-0.3.20-1
mingw-w64-x86_64-opencore-amr-0.1.5-1 mingw-w64-x86_64-opencv-4.5.2-4
mingw-w64-x86_64-openexr-2.5.7-6 mingw-w64-x86_64-openjpeg2-2.4.0-3
mingw-w64-x86_64-opus-1.3.1-4 mingw-w64-x86_64-protobuf-3.16.0-3
mingw-w64-x86_64-pugixml-1.12-2 mingw-w64-x86_64-rav1e-0.4.1-1
mingw-w64-x86_64-rtmpdump-git-r514.c5f04a5-4 mingw-w64-x86_64-speex-1.2.0-1
mingw-w64-x86_64-speexdsp-1.2.0-1 mingw-w64-x86_64-srt-1.4.4-1
mingw-w64-x86_64-suitesparse-5.10.1-2 mingw-w64-x86_64-svt-av1-0.9.1-1
mingw-w64-x86_64-tesseract-ocr-5.1.0-1 mingw-w64-x86_64-tinyxml-2.6.2-5
mingw-w64-x86_64-vid.stab-1.1-1 mingw-w64-x86_64-vulkan-headers-1.3.207-1
mingw-w64-x86_64-vulkan-loader-1.3.207-1
mingw-w64-x86_64-winpthreads-git-9.0.0.6448.b03cbfb95-1
mingw-w64-x86_64-x264-0.161.r3039.544c61f-2 mingw-w64-x86_64-x265-3.5-1
mingw-w64-x86_64-xvidcore-1.3.7-2 mingw-w64-x86_64-zimg-3.0.3-1
mingw-w64-x86_64-zziplib-0.13.72-3 mingw-w64-x86_64-ffmpeg-4.4.1-11
Total Download Size: 0.63 MiB
Total Installed Size: 1551.40 MiB
:: Proceed with installation? [Y/n]
:: Retrieving packages...
mingw-w64-x86_64-ffms2-2.40-1-any downloading...
mingw-w64-x86_64-libvorbis-1.3.7-1-any downloading...
mingw-w64-x86_64-lz4-1.9.3-1-any downloading...
checking keyring...
checking package integrity...
loading package files...
checking for file conflicts...
checking available disk space...
:: Processing package changes...
installing mingw-w64-x86_64-aom...
installing mingw-w64-x86_64-libogg...
installing mingw-w64-x86_64-celt...
installing mingw-w64-x86_64-gavl...
installing mingw-w64-x86_64-eigen3...
installing mingw-w64-x86_64-gflags...
installing mingw-w64-x86_64-libunwind...
installing mingw-w64-x86_64-glog...
installing mingw-w64-x86_64-gcc-libgfortran...
installing mingw-w64-x86_64-openblas...
Set the environment variable OPENBLAS_NUM_THREADS to the
number of threads to use.
installing mingw-w64-x86_64-metis...
installing mingw-w64-x86_64-suitesparse...
installing mingw-w64-x86_64-ceres-solver...
installing mingw-w64-x86_64-intel-tbb...
installing mingw-w64-x86_64-freeglut...
installing mingw-w64-x86_64-jasper...
installing mingw-w64-x86_64-libaec...
installing mingw-w64-x86_64-hdf5...
installing mingw-w64-x86_64-icu...
installing mingw-w64-x86_64-boost...
mingw-w64-x86_64-python-numpy: For Boost.Python (NumPy)
mingw-w64-x86_64-python: For Boost.Python [installed]
Optional dependencies for mingw-w64-x86_64-boost
installing mingw-w64-x86_64-cppunit...
installing mingw-w64-x86_64-jxrlib...
installing mingw-w64-x86_64-lcms2...
installing mingw-w64-x86_64-libraw...
installing mingw-w64-x86_64-openjpeg2...
installing mingw-w64-x86_64-ilmbase...
installing mingw-w64-x86_64-openexr...
installing mingw-w64-x86_64-freeimage...
installing mingw-w64-x86_64-glsl-optimizer-git...
installing mingw-w64-x86_64-hlsl2glsl-git...
installing mingw-w64-x86_64-vulkan-headers...
installing mingw-w64-x86_64-vulkan-loader...
installing mingw-w64-x86_64-SDL2...
installing mingw-w64-x86_64-pugixml...
installing mingw-w64-x86_64-tinyxml...
installing mingw-w64-x86_64-headers-git...
installing mingw-w64-x86_64-crt-git...
installing mingw-w64-x86_64-winpthreads-git...
installing mingw-w64-x86_64-zziplib...
mingw-w64-x86_64-SDL: SDL_rwops for ZZipLib
Optional dependencies for mingw-w64-x86_64-zziplib
installing mingw-w64-x86_64-ogre3d...
installing mingw-w64-x86_64-protobuf...
installing mingw-w64-x86_64-libgpg-error...
installing mingw-w64-x86_64-libgcrypt...
installing mingw-w64-x86_64-libxslt...
installing mingw-w64-x86_64-docbook-xml...
installing mingw-w64-x86_64-docbook-xsl...
installing mingw-w64-x86_64-asciidoc...
installing mingw-w64-x86_64-libunistring...
installing mingw-w64-x86_64-libidn2...
installing mingw-w64-x86_64-libpsl...
installing mingw-w64-x86_64-jansson...
installing mingw-w64-x86_64-jemalloc...
installing mingw-w64-x86_64-nghttp2...
installing mingw-w64-x86_64-curl...
installing mingw-w64-x86_64-leptonica...
installing mingw-w64-x86_64-libb2...
installing mingw-w64-x86_64-lz4...
installing mingw-w64-x86_64-nettle...
installing mingw-w64-x86_64-libarchive...
installing mingw-w64-x86_64-tesseract-ocr...
installing mingw-w64-x86_64-opencv...
mingw-w64-x86_64-ffmpeg: support to read and write video files [pending]
mingw-w64-x86_64-python-numpy: Python 3.x interface
Optional dependencies for mingw-w64-x86_64-opencv
installing mingw-w64-x86_64-frei0r-plugins...
installing mingw-w64-x86_64-dav1d...
installing mingw-w64-x86_64-gnutls...
installing mingw-w64-x86_64-gsm...
installing mingw-w64-x86_64-lame...
installing mingw-w64-x86_64-libass...
installing mingw-w64-x86_64-libbluray...
installing mingw-w64-x86_64-libcaca...
installing mingw-w64-x86_64-libexif...
installing mingw-w64-x86_64-libgme...
installing mingw-w64-x86_64-libmfx...
installing mingw-w64-x86_64-libmodplug...
installing mingw-w64-x86_64-libssh...
installing mingw-w64-x86_64-libsoxr...
installing mingw-w64-x86_64-libvorbis...
installing mingw-w64-x86_64-libtheora...
installing mingw-w64-x86_64-libvpx...
installing mingw-w64-x86_64-libmysofa...
installing mingw-w64-x86_64-openal...
installing mingw-w64-x86_64-opencore-amr...
installing mingw-w64-x86_64-opus...
installing mingw-w64-x86_64-rav1e...
installing mingw-w64-x86_64-rtmpdump-git...
installing mingw-w64-x86_64-speexdsp...
installing mingw-w64-x86_64-speex...
installing mingw-w64-x86_64-srt...
installing mingw-w64-x86_64-svt-av1...
installing mingw-w64-x86_64-vid.stab...
installing mingw-w64-x86_64-l-smash...
installing mingw-w64-x86_64-ffms2...
installing mingw-w64-x86_64-x264...
installing mingw-w64-x86_64-x265...
installing mingw-w64-x86_64-xvidcore...
installing mingw-w64-x86_64-zimg...
installing mingw-w64-x86_64-ffmpeg...
Update operation finished
Installation complete
```

</p>
</details>

**Comment by axcore** [comment]
Both problems will probably resolve themselves in a few days, and then everything will be back to normal.

**Comment by goodstuff626** [comment]
Is this problem still not fixed? I just installed tartube v2.4.077, for windows 10, and ffmpeg is still failing to install with the default settings. Retried 4 times, all failed. I had to edit the pacman.conf to get it to install.

**Comment by eenturk** [comment]
> Is this problem still not fixed? I just installed tartube v2.4.077, for windows 10, and ffmpeg is still failing to install with the default settings. Retried 4 times, all failed. I had to edit the pacman.conf to get it to install.

Tried on a fresh machine still same issue. Seems like something that can be easy fixed but of course i don't know what is going on in the back end. Also the pacman fix seems to only work when it's a fresh install. Tried this fix on a existing install and still failing.

**Comment by heywolf15** [comment]
@goodstuff626 @eenturk  Try completely delete Tartube folder and reinstall it

**Comment by eenturk** [comment]
> @goodstuff626 @eenturk Try completely delete Tartube folder and reinstall it

Yeah thats what i did and that worked out fine and only works when you change it after it installed, changed the pacman file and then start up Tartube for the first time when it also wants you to install ffmpeg.

**Comment by axcore** [comment]
> FFmpeg install fails with `msys2-x86_64-20220319.exe` (the version that Tartube installers are using), but succeeds with `msys2-x86_64-20220603.exe`.
> 
> However, I can't update to the latest version of MSYS2, because yt-dlp is not installable on it (because `pip3 install brotli` fails).

The brotli problem has resolved itself and FFmpeg is now installable again. I will upload a new release over this weekend.

**Comment by axcore** [comment]
v2.4.093 has been released, with the latest version of MSYS2, with which FFmpeg can be downloaded as normal.

I have also released an installer for MS Windows that includes FFMpeg, but it's a very large download (about 800MB). Perhaps you can try it out, and tell me what you think. In the setup window, when you're asked to install FFmpeg, just skip past that page.

The file is [here](https://github.com/axcore/tartube/releases/download/v2.4.093/install-tartube-2.4.093-64bit-ffmpeg.exe).

**Comment by comiluv** [comment]
Installation fails on v2.4.093 on Windows 11. Installed tartube via Chocolatey. Below is the log.

<details>
<summary>FFmpeg installation fails</summary>
업데이트 작업 시작, FFmpeg 설치중
pacman -S mingw-w64-x86_64-ffmpeg --noconfirm
resolving dependencies...
looking for conflicting packages...



Packages (95) mingw-w64-x86_64-SDL2-2.0.22-2 mingw-w64-x86_64-aom-3.4.0-1
mingw-w64-x86_64-asciidoc-10.2.0-2 mingw-w64-x86_64-boost-1.79.0-4
mingw-w64-x86_64-ceres-solver-2.1.0-2 mingw-w64-x86_64-cppunit-1.15.1-2
mingw-w64-x86_64-crt-git-10.0.0.r54.gb4116e310-1 mingw-w64-x86_64-curl-7.84.0-2
mingw-w64-x86_64-dav1d-1.0.0-1 mingw-w64-x86_64-docbook-xml-1~4.5-3
mingw-w64-x86_64-docbook-xsl-1.79.2-8 mingw-w64-x86_64-eigen3-3.4.0-1
mingw-w64-x86_64-ffms2-2.40-1 mingw-w64-x86_64-freeglut-3.2.2-2
mingw-w64-x86_64-freeimage-3.18.0-8 mingw-w64-x86_64-frei0r-plugins-1.8.0-1
mingw-w64-x86_64-gavl-1.4.0-1 mingw-w64-x86_64-gcc-libgfortran-12.1.0-3
mingw-w64-x86_64-gflags-2.2.2-4 mingw-w64-x86_64-glog-0.6.0-1
mingw-w64-x86_64-glsl-optimizer-git-1~r66913.d78c3d2f24-1
mingw-w64-x86_64-gnutls-3.7.6-2 mingw-w64-x86_64-gsm-1.0.19-2
mingw-w64-x86_64-hdf5-1.12.2-1
mingw-w64-x86_64-headers-git-10.0.0.r54.gb4116e310-1
mingw-w64-x86_64-hlsl2glsl-git-1~r848.14dd189-1 mingw-w64-x86_64-icu-69.1-1
mingw-w64-x86_64-ilmbase-2.5.8-3 mingw-w64-x86_64-intel-tbb-1~2020.3-1
mingw-w64-x86_64-jansson-2.14-2 mingw-w64-x86_64-jasper-3.0.6-1
mingw-w64-x86_64-jemalloc-5.2.1-2 mingw-w64-x86_64-jxrlib-1.1-3
mingw-w64-x86_64-l-smash-2.14.5-3 mingw-w64-x86_64-lame-3.100-2
mingw-w64-x86_64-lcms2-2.13.1-2 mingw-w64-x86_64-leptonica-1.82.0-1
mingw-w64-x86_64-libaec-1.0.6-2 mingw-w64-x86_64-libarchive-3.6.1-2
mingw-w64-x86_64-libass-0.16.0-1 mingw-w64-x86_64-libb2-0.98.1-2
mingw-w64-x86_64-libbluray-1.3.1-1 mingw-w64-x86_64-libcaca-0.99.beta20-3
mingw-w64-x86_64-libexif-0.6.24-2 mingw-w64-x86_64-libgcrypt-1.10.1-1
mingw-w64-x86_64-libgme-0.6.3-3 mingw-w64-x86_64-libgpg-error-1.45-1
mingw-w64-x86_64-libidn2-2.3.3-1 mingw-w64-x86_64-libmfx-1.35.1-1
mingw-w64-x86_64-libmodplug-0.8.9.0-4 mingw-w64-x86_64-libmysofa-1.2.1-1
mingw-w64-x86_64-libogg-1.3.5-1 mingw-w64-x86_64-libpsl-0.21.1-4
mingw-w64-x86_64-libraw-0.20.2-6 mingw-w64-x86_64-libsoxr-0.1.3-4
mingw-w64-x86_64-libssh-0.9.6-2 mingw-w64-x86_64-libtheora-1.1.1-7
mingw-w64-x86_64-libunistring-1.0-1 mingw-w64-x86_64-libunwind-14.0.4-2
mingw-w64-x86_64-libvorbis-1.3.7-1 mingw-w64-x86_64-libvpx-1.12.0-1
mingw-w64-x86_64-libxslt-1.1.35-1 mingw-w64-x86_64-lz4-1.9.3-1
mingw-w64-x86_64-metis-5.1.0-4 mingw-w64-x86_64-nettle-3.8-1
mingw-w64-x86_64-nghttp2-1.48.0-1 mingw-w64-x86_64-ogre3d-1.12.6-5
mingw-w64-x86_64-openal-1.22.2-1 mingw-w64-x86_64-openblas-0.3.20-3
mingw-w64-x86_64-opencore-amr-0.1.5-2 mingw-w64-x86_64-opencv-4.5.5-3
mingw-w64-x86_64-openexr-2.5.8-3 mingw-w64-x86_64-openjpeg2-2.5.0-2
mingw-w64-x86_64-opus-1.3.1-4 mingw-w64-x86_64-protobuf-3.16.0-3
mingw-w64-x86_64-pugixml-1.12.1-1 mingw-w64-x86_64-rav1e-0.5.1-3
mingw-w64-x86_64-rtmpdump-git-r514.c5f04a5-4 mingw-w64-x86_64-speex-1.2.1-1
mingw-w64-x86_64-speexdsp-1.2.1-1 mingw-w64-x86_64-srt-1.5.0-1
mingw-w64-x86_64-suitesparse-5.12.0-2 mingw-w64-x86_64-svt-av1-1.1.0-1
mingw-w64-x86_64-tesseract-ocr-5.2.0-1 mingw-w64-x86_64-tinyxml-2.6.2-5
mingw-w64-x86_64-vid.stab-1.1-2 mingw-w64-x86_64-vulkan-headers-1.3.222-1
mingw-w64-x86_64-vulkan-loader-1.3.221-1
mingw-w64-x86_64-winpthreads-git-10.0.0.r54.gb4116e310-1
mingw-w64-x86_64-x264-0.164.r3094.bfc87b7-1 mingw-w64-x86_64-x265-3.5-3
mingw-w64-x86_64-xvidcore-1.3.7-3 mingw-w64-x86_64-zimg-3.0.4-1
mingw-w64-x86_64-zziplib-0.13.72-3 mingw-w64-x86_64-ffmpeg-4.4.2-4
Total Download Size: 272.69 MiB
Total Installed Size: 1415.13 MiB
:: Proceed with installation? [Y/n]
:: Retrieving packages...
mingw-w64-x86_64-SDL2-2.0.22-2-any downloading...
mingw-w64-x86_64-aom-3.4.0-1-any downloading...
mingw-w64-x86_64-asciidoc-10.2.0-2-any downloading...
mingw-w64-x86_64-boost-1.79.0-4-any downloading...
mingw-w64-x86_64-ceres-solver-2.1.0-2-any downloading...
mingw-w64-x86_64-cppunit-1.15.1-2-any downloading...
mingw-w64-x86_64-crt-git-10.0.0.r54.gb4116e310-1-any downloading...
mingw-w64-x86_64-curl-7.84.0-2-any downloading...
mingw-w64-x86_64-dav1d-1.0.0-1-any downloading...
mingw-w64-x86_64-docbook-xml-1~4.5-3-any downloading...
mingw-w64-x86_64-docbook-xsl-1.79.2-8-any downloading...
mingw-w64-x86_64-eigen3-3.4.0-1-any downloading...
mingw-w64-x86_64-ffmpeg-4.4.2-4-any downloading...
mingw-w64-x86_64-ffms2-2.40-1-any downloading...
mingw-w64-x86_64-freeglut-3.2.2-2-any downloading...
mingw-w64-x86_64-freeimage-3.18.0-8-any downloading...
mingw-w64-x86_64-frei0r-plugins-1.8.0-1-any downloading...
mingw-w64-x86_64-gavl-1.4.0-1-any downloading...
mingw-w64-x86_64-gcc-libgfortran-12.1.0-3-any downloading...
mingw-w64-x86_64-gflags-2.2.2-4-any downloading...
mingw-w64-x86_64-glog-0.6.0-1-any downloading...
mingw-w64-x86_64-glsl-optimizer-git-1~r66913.d78c3d2f24-1-any downloading...
mingw-w64-x86_64-gnutls-3.7.6-2-any downloading...
mingw-w64-x86_64-gsm-1.0.19-2-any downloading...
mingw-w64-x86_64-hdf5-1.12.2-1-any downloading...
mingw-w64-x86_64-headers-git-10.0.0.r54.gb4116e310-1-any downloading...
mingw-w64-x86_64-hlsl2glsl-git-1~r848.14dd189-1-any downloading...
mingw-w64-x86_64-icu-69.1-1-any downloading...
mingw-w64-x86_64-ilmbase-2.5.8-3-any downloading...
mingw-w64-x86_64-intel-tbb-1~2020.3-1-any downloading...
mingw-w64-x86_64-jansson-2.14-2-any downloading...
mingw-w64-x86_64-jasper-3.0.6-1-any downloading...
mingw-w64-x86_64-jemalloc-5.2.1-2-any downloading...
mingw-w64-x86_64-jxrlib-1.1-3-any downloading...
mingw-w64-x86_64-l-smash-2.14.5-3-any downloading...
mingw-w64-x86_64-lame-3.100-2-any downloading...
mingw-w64-x86_64-lcms2-2.13.1-2-any downloading...
mingw-w64-x86_64-leptonica-1.82.0-1-any downloading...
mingw-w64-x86_64-libaec-1.0.6-2-any downloading...
mingw-w64-x86_64-libarchive-3.6.1-2-any downloading...
mingw-w64-x86_64-libass-0.16.0-1-any downloading...
mingw-w64-x86_64-libb2-0.98.1-2-any downloading...
mingw-w64-x86_64-libbluray-1.3.1-1-any downloading...
mingw-w64-x86_64-libcaca-0.99.beta20-3-any downloading...
mingw-w64-x86_64-libexif-0.6.24-2-any downloading...
mingw-w64-x86_64-libgcrypt-1.10.1-1-any downloading...
mingw-w64-x86_64-libgme-0.6.3-3-any downloading...
mingw-w64-x86_64-libgpg-error-1.45-1-any downloading...
mingw-w64-x86_64-libidn2-2.3.3-1-any downloading...
mingw-w64-x86_64-libmfx-1.35.1-1-any downloading...
mingw-w64-x86_64-libmodplug-0.8.9.0-4-any downloading...
mingw-w64-x86_64-libmysofa-1.2.1-1-any downloading...
mingw-w64-x86_64-libogg-1.3.5-1-any downloading...
mingw-w64-x86_64-libpsl-0.21.1-4-any downloading...
mingw-w64-x86_64-libraw-0.20.2-6-any downloading...
mingw-w64-x86_64-libsoxr-0.1.3-4-any downloading...
mingw-w64-x86_64-libssh-0.9.6-2-any downloading...
mingw-w64-x86_64-libtheora-1.1.1-7-any downloading...
mingw-w64-x86_64-libunistring-1.0-1-any downloading...
mingw-w64-x86_64-libunwind-14.0.4-2-any downloading...
mingw-w64-x86_64-libvorbis-1.3.7-1-any downloading...
mingw-w64-x86_64-libvpx-1.12.0-1-any downloading...
mingw-w64-x86_64-libxslt-1.1.35-1-any downloading...
mingw-w64-x86_64-lz4-1.9.3-1-any downloading...
mingw-w64-x86_64-metis-5.1.0-4-any downloading...
mingw-w64-x86_64-nettle-3.8-1-any downloading...
mingw-w64-x86_64-nghttp2-1.48.0-1-any downloading...
mingw-w64-x86_64-ogre3d-1.12.6-5-any downloading...
mingw-w64-x86_64-openal-1.22.2-1-any downloading...
mingw-w64-x86_64-openblas-0.3.20-3-any downloading...
mingw-w64-x86_64-opencore-amr-0.1.5-2-any downloading...
mingw-w64-x86_64-opencv-4.5.5-3-any downloading...
mingw-w64-x86_64-openexr-2.5.8-3-any downloading...
mingw-w64-x86_64-openjpeg2-2.5.0-2-any downloading...
mingw-w64-x86_64-opus-1.3.1-4-any downloading...
mingw-w64-x86_64-protobuf-3.16.0-3-any downloading...
mingw-w64-x86_64-pugixml-1.12.1-1-any downloading...
mingw-w64-x86_64-rav1e-0.5.1-3-any downloading...
mingw-w64-x86_64-rtmpdump-git-r514.c5f04a5-4-any downloading...
mingw-w64-x86_64-speex-1.2.1-1-any downloading...
mingw-w64-x86_64-speexdsp-1.2.1-1-any downloading...
mingw-w64-x86_64-srt-1.5.0-1-any downloading...
mingw-w64-x86_64-suitesparse-5.12.0-2-any downloading...
mingw-w64-x86_64-svt-av1-1.1.0-1-any downloading...
mingw-w64-x86_64-tesseract-ocr-5.2.0-1-any downloading...
mingw-w64-x86_64-tinyxml-2.6.2-5-any downloading...
mingw-w64-x86_64-vid.stab-1.1-2-any downloading...
mingw-w64-x86_64-vulkan-headers-1.3.222-1-any downloading...
mingw-w64-x86_64-vulkan-loader-1.3.221-1-any downloading...
mingw-w64-x86_64-winpthreads-git-10.0.0.r54.gb4116e310-1-any downloading...
mingw-w64-x86_64-x264-0.164.r3094.bfc87b7-1-any downloading...
mingw-w64-x86_64-x265-3.5-3-any downloading...
mingw-w64-x86_64-xvidcore-1.3.7-3-any downloading...
mingw-w64-x86_64-zimg-3.0.4-1-any downloading...
mingw-w64-x86_64-zziplib-0.13.72-3-any downloading...
checking keyring...
checking package integrity...
error: mingw-w64-x86_64-lz4: signature from "David Macek
<david.macek.0@gmail.com>" is unknown trust
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-lz4-1.9.3-1-any.pkg.tar.zst is
corrupted (invalid or corrupted package (PGP signature)).
error: mingw-w64-x86_64-libvorbis: signature from "David Macek
<david.macek.0@gmail.com>" is unknown trust
Do you want to delete it? [Y/n]
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-libvorbis-1.3.7-1-any.pkg.tar.zst
is corrupted (invalid or corrupted package (PGP signature)).
error: mingw-w64-x86_64-ffms2: signature from "David Macek
<david.macek.0@gmail.com>" is unknown trust
:: File /var/cache/pacman/pkg/mingw-w64-x86_64-ffms2-2.40-1-any.pkg.tar.zst is
corrupted (invalid or corrupted package (PGP signature)).
Do you want to delete it? [Y/n]
error: failed to commit transaction (invalid or corrupted package)
Do you want to delete it? [Y/n]
Errors occurred, no packages were upgraded.
업데이트 작업이 끝남
Installation failed



</details>

**Comment by kurochicken** [comment]
Windows 10. 2.4.165 and 2.4.221. Also can't install FFmpeg, has an error. Trying to run as admin, and it has no affect:

```
Starting update operation, installing FFmpeg
pacman -S mingw-w64-x86_64-ffmpeg --noconfirm
resolving dependencies...
looking for conflicting packages...



:: Proceed with installation? [Y/n]
:: Retrieving packages...
Packages (91) mingw-w64-x86_64-SDL2-2.26.0-1 mingw-w64-x86_64-aom-3.5.0-1
mingw-w64-x86_64-asciidoc-10.2.0-2 mingw-w64-x86_64-boost-1.80.0-2
mingw-w64-x86_64-bullet-3.24-3 mingw-w64-x86_64-ceres-solver-2.1.0-2
mingw-w64-x86_64-crt-git-10.0.0.r130.g13f9449c1-1 mingw-w64-x86_64-curl-7.86.0-5
mingw-w64-x86_64-dav1d-1.0.0-1 mingw-w64-x86_64-docbook-xml-1~4.5-3
mingw-w64-x86_64-docbook-xsl-1.79.2-8 mingw-w64-x86_64-eigen3-3.4.0-1
mingw-w64-x86_64-ffms2-2.40-1 mingw-w64-x86_64-freeglut-3.4.0-2
mingw-w64-x86_64-frei0r-plugins-1.8.0-2 mingw-w64-x86_64-gavl-1.4.0-1
mingw-w64-x86_64-gcc-libgfortran-12.2.0-6 mingw-w64-x86_64-gflags-2.2.2-4
mingw-w64-x86_64-glog-0.6.0-1
mingw-w64-x86_64-glsl-optimizer-git-1~r66913.d78c3d2f24-1
mingw-w64-x86_64-gnutls-3.7.8-1 mingw-w64-x86_64-gsm-1.0.22-1
mingw-w64-x86_64-hdf5-1.12.2-1
mingw-w64-x86_64-headers-git-10.0.0.r130.g13f9449c1-1
mingw-w64-x86_64-hlsl2glsl-git-1~r848.14dd189-1 mingw-w64-x86_64-hwloc-2.8.0-1
mingw-w64-x86_64-icu-72.1-1 mingw-w64-x86_64-imath-3.1.6-1
mingw-w64-x86_64-jasper-4.0.0-1 mingw-w64-x86_64-l-smash-2.14.5-3
mingw-w64-x86_64-lame-3.100-2 mingw-w64-x86_64-lcms2-2.14-1
mingw-w64-x86_64-leptonica-1.82.0-1 mingw-w64-x86_64-libaec-1.0.6-2
mingw-w64-x86_64-libarchive-3.6.1-2 mingw-w64-x86_64-libass-0.16.0-1
mingw-w64-x86_64-libb2-0.98.1-2 mingw-w64-x86_64-libbluray-1.3.3-1
mingw-w64-x86_64-libcaca-0.99.beta20-3 mingw-w64-x86_64-libexif-0.6.24-2
mingw-w64-x86_64-libgcrypt-1.10.1-1 mingw-w64-x86_64-libgme-0.6.3-3
mingw-w64-x86_64-libgpg-error-1.46-2 mingw-w64-x86_64-libidn2-2.3.3-1
mingw-w64-x86_64-libltdl-2.4.7-1 mingw-w64-x86_64-libmfx-1.35.1-1
mingw-w64-x86_64-libmodplug-0.8.9.0-4 mingw-w64-x86_64-libmysofa-1.3.1-1
mingw-w64-x86_64-libogg-1.3.5-1 mingw-w64-x86_64-libpsl-0.21.1-4
mingw-w64-x86_64-libsoxr-0.1.3-4 mingw-w64-x86_64-libssh-0.10.4-1
mingw-w64-x86_64-libtheora-1.1.1-7 mingw-w64-x86_64-libunistring-1.0-1
mingw-w64-x86_64-libunwind-15.0.4-1 mingw-w64-x86_64-libvorbis-1.3.7-1
mingw-w64-x86_64-libvpx-1.12.0-1 mingw-w64-x86_64-libxslt-1.1.37-1
mingw-w64-x86_64-lz4-1.9.4-1 mingw-w64-x86_64-metis-5.1.0-4
mingw-w64-x86_64-nettle-3.8.1-1 mingw-w64-x86_64-nghttp2-1.51.0-1
mingw-w64-x86_64-ogre3d-13.5.2-1 mingw-w64-x86_64-openal-1.22.2-1
mingw-w64-x86_64-openblas-0.3.21-6 mingw-w64-x86_64-opencore-amr-0.1.6-1
mingw-w64-x86_64-opencv-4.6.0-9 mingw-w64-x86_64-openexr-3.1.5-2
mingw-w64-x86_64-openjpeg2-2.5.0-2 mingw-w64-x86_64-openvr-1.23.7-1
mingw-w64-x86_64-opus-1.3.1-5 mingw-w64-x86_64-protobuf-21.9-1
mingw-w64-x86_64-rav1e-0.5.1-3 mingw-w64-x86_64-rtmpdump-git-r514.c5f04a5-4
mingw-w64-x86_64-speex-1.2.1-1 mingw-w64-x86_64-speexdsp-1.2.1-1
mingw-w64-x86_64-srt-1.5.1-1 mingw-w64-x86_64-suitesparse-5.13.0-1
mingw-w64-x86_64-svt-av1-1.3.0-1 mingw-w64-x86_64-tbb-2021.7.0-2
mingw-w64-x86_64-tesseract-ocr-5.2.0-3 mingw-w64-x86_64-tinyxml-2.6.2-5
mingw-w64-x86_64-vid.stab-1.1-2 mingw-w64-x86_64-vulkan-headers-1.3.235-1
mingw-w64-x86_64-vulkan-loader-1.3.235-1
mingw-w64-x86_64-winpthreads-git-10.0.0.r130.g13f9449c1-1
mingw-w64-x86_64-x264-0.164.r3094.bfc87b7-1 mingw-w64-x86_64-x265-3.5-3
mingw-w64-x86_64-xvidcore-1.3.7-3 mingw-w64-x86_64-zimg-3.0.4-1
mingw-w64-x86_64-ffmpeg-4.4.3-3
Total Download Size: 279.11 MiB
Total Installed Size: 1644.78 MiB
Update operation finished
Installation complete
Starting update operation, installing FFmpeg
pacman -S mingw-w64-x86_64-ffmpeg --noconfirm
if you're sure a package manager is not already
running, you can remove /var/lib/pacman/db.lck
error: could not lock database: File exists
error: failed to init transaction (unable to lock database)
Update operation finished
Installation failed
```

**Comment by axcore** [comment]
I am not very familiar with MSYS2 packaging so I can't explain why you are seeing this error. 

I tested FFmpeg installation on Windows 10 today, and it is working as expected. Sometimes the software repository is broken for a short time.

You should try **completely deleting** Tartube's installation folder before re-installing it, because the message `Starting update operation, installing FFmpeg` implies that pacman is trying to repair a broken FFmpeg installation but failing.

**Comment by axcore** [comment]
Closing this issue again, as FFmpeg seems to be working fine. Feel free to open a new issue, if it breaks again.
